src/mongoc/mongoc-topology-description.c

```diff
@@ -445,7 +445,13 @@ mongoc_topology_description_select (mongoc_topology_description_t *topology,
    }
 
    if (topology->type == MONGOC_TOPOLOGY_SINGLE) {
-      return topology->servers->items[0].item;
+      sd = topology->servers->items[0].item;
```

Comment:
I think it's better for server selection to say we don't have a server if we've never managed to ismaster it at all